src/wp-includes/interactivity-api/class-wp-interactivity-api.php

```diff
@@ -208,6 +208,9 @@ public function register_script_modules() {
 	public function add_hooks() {
 		add_action( 'wp_enqueue_scripts', array( $this, 'register_script_modules' ) );
 		add_action( 'wp_footer', array( $this, 'print_client_interactivity_data' ) );
+
+		add_action( 'admin_enqueue_scripts', array( $this, 'register_script_modules' ) );
```

Comment:
The question is which hook fits better here as there are two possibilities in contrast to the front end:
- `admin_footer` - Prints scripts or data before the default footer scripts.
- `admin_print_footer_scripts` - Prints any scripts and data queued for the footer.

Reply:
It's the configs and initial state for stores so it probably needs to be printer earlier 👍🏻

Reply:
Oh, good question! I think I'd print this as late as possible. `@wordpress/interactivity` is what will use this when it initializes, and because that's a module the DOM should have finished parsing before it runs.

I think we can switch this to `admin_print_footer_scripts` to print it as late as possible.

Reply:
I was exploring that part of the codebase for the first time today. I still don't know what should be the order of printing when taking into account also 3rd party plugins that depend on Interactivity API. My initial thinking was that  3rd party plugins would use `admin_enqueue_scripts` to register/enqueue their scripts. I'm not sure how folks would use `wp_interactivity_store`.

Reply:
I've pushed that change, I can see the script moved after the footer scripts and it continues to work as expected.

Reply:
It's `@wordpress/interactivity` itself that [reads data from the printed store data](https://github.com/WordPress/gutenberg/blob/1eb76440f7b86ca905ac875d4a2c6ed67fc48daf/packages/interactivity/src/store.ts#L322-L334). And because these are modules (which implies defer) there shouldn't be any problem pushing the printed script tag to be later. The module that depends on it will not execute until the DOM has been parsed completely, so the script tag it looks for will be there.